drivers/apc_modbus.c

```diff
@@ -0,0 +1,1229 @@
+/*  apc_modbus.c - Driver for APC Modbus UPS
+ *  Copyright © 2023 Axel Gembe <axel@gembe.net>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include "main.h"
+# if defined NUT_MODBUS_HAS_USB
+#include "nut_libusb.h"
+#include "libhid.h"
+#include "hidparser.h"
+# endif /* defined NUT_MODBUS_HAS_USB */
```

Comment:
Preprocessing indentation looks funny :)

Not a big deal, and not easy to maintain consistently in the long run (especially for copy-pasted warnings pragma indentations), but in most places I've tried to keep the outer-most directives flushed to the left, and things inside an `#if(def)` block indented as they get nested, to better see what is optional when.

No rush to change this here (and below) now, but then expect that it might get groomed differently in NUT codebase eventually :)